Add self-serve SSO docs (#3349)
clerkinc/clerk-docs: docs/reference/components/authentication/configure-sso.mdx

---
title: '`<ConfigureSSO />` component'
description: Clerk's <ConfigureSSO /> component renders a self-serve UI that lets your customers' IT admins configure their own enterprise SSO connections from inside your application.
sdk: astro, nextjs, react, tanstack-react-start, vue, js-frontend
---

> The `<ConfigureSSO />` component renders a UI with instructions that allows end-users, typically the IT admin at one of your enterprise customers, configure single sign-on directly from your application.

Reviewer: ```suggestion
The `<ConfigureSSO />` component renders a UI with instructions that allows end-users, typically the IT admin for one of your enterprise customers, to configure single sign-on directly from your application.
```

> The `<ConfigureSSO />` component renders a guided wizard that lets a signed-in user — typically the IT admin at one of your enterprise customers — configure an enterprise SSO connection end-to-end without leaving your application or touching the Clerk Dashboard.

Reviewer 2: ```suggestion
The `<ConfigureSSO />` component renders a UI with instructions that allows end-users, typically the IT admin at one of your enterprise customers, configure single sign-on directly from your application.
```

Couple of points: 
- Let's avoid mentioning "guided wizard", it seems too specific to the UI pattern / design term, if one day we end up removing the wizard from the UI
- Removing signed-in user, for Admin Portal we'll allow "sessionless" - we could just mention end-user here
- "touching the Clerk Dashboard" -> this is from the C1 perspective

> The `<ConfigureSSO />` component renders a guided wizard that lets a signed-in user — typically the IT admin at one of your enterprise customers — configure an enterprise SSO connection end-to-end without leaving your application or touching the Clerk Dashboard.

Reviewer 2: Let's add an screenshot of the component once we release it

> The wizard walks the user through selecting an Identity Provider (IdP), verifying their email domain, supplying the protocol-specific configuration (with IdP setup instructions embedded inline), running a test sign-in, and activating the connection.

Reviewer: we've never used "wizard" in the docs, we usually use "flow" or "process" 💭

Reviewer: updated! [6232912](https://github.com/clerk/clerk-docs/pull/3349/commits/6232912c72277cfc2015baccb0aa9e0261fb3667)

> `When Organizations are enabled, the connection is attached to the user's active Organization, otherwise it's attached to the email domain at the instance level.

Reviewer: ```suggestion
When Organizations are enabled, the connection is attached to the user's [Active Organization](!active-organization). Otherwise, it's attached to the email domain at the instance level.
```

> `<ConfigureSSO />` works whether or not your application uses [Clerk Organizations](/docs/guides/organizations/overview). When Organizations are enabled, the connection is attached to the user's active Organization; otherwise it's attached to the email domain at the application level.

Reviewer 2: ```suggestion
`When Organizations are enabled, the connection is attached to the user's active Organization, otherwise it's attached to the email domain at the instance level.
```

A bit more concize

To learn how the broader self-serve SSO feature works and how to enable it in the Clerk Dashboard, see the [self-serve SSO guide](/docs/guides/configure/auth-strategies/enterprise-connections/self-serve-sso).

> [!NOTE]
> In its current implementation, `<ConfigureSSO />` is the only way to use self-serve SSO, and it requires an active Clerk session. The IT admin must have a Clerk user account in your application and be signed in to use the flow. It currently only supports Okta and custom SAML.

## Requirements

`<ConfigureSSO />` will only render when all of the following are true:

- Self-serve SSO is enabled for your instance. See the [self-serve SSO guide](/docs/guides/configure/auth-strategies/enterprise-connections/self-serve-sso#enable-self-serve-sso) for instructions.
- The **Email address** identifier is enabled on your instance.

> - The component is rendered to a signed-in user. The component reads the active Clerk session and does not currently support unauthenticated, token-based access.

Reviewer 2: ```suggestion
- The component is rendered to a signed-in user. 
```

I think it's not needed to mention token-based access here

If any of these conditions are not met, the component will throw a runtime error in development and will not render in production.

> If you use Clerk Organizations, the signed-in user must also have the `org:sys_enterprise_connections` system permission in their active Organization to manage enterprise connections. The permission ships with the default **Admin** role.

Reviewer: ```suggestion
If you use Clerk Organizations, the signed-in user must also have the `org:sys_enterprise_connections` system permission in their [Active Organization](!active-organization) to manage enterprise connections. The permission ships with the default **Admin** role.
```

<If notSdk="js-frontend">
  ## Example

  The following example includes a basic implementation of the `<ConfigureSSO />` component. You can use this as a starting point for your own implementation.

>   Because `<ConfigureSSO />` is experimental, it must be imported from the `experimental` subpath of your SDK.

Reviewer 3: @im2nguyen If it's experimental, we may want to add a callout at the top of the file, like this one:

<img width="881" height="80" alt="Image" src="https://github.com/user-attachments/assets/b24ea635-4cee-451d-9003-4a96ae08bd95" />

Or:

<img width="873" height="131" alt="Image" src="https://github.com/user-attachments/assets/bc1cbef1-d5ee-4761-8d4c-071deeeea173" />

Author: thanks! we need to update this since we're launching with ss-sso through orgprofile first. its only experimental now, but when this goes live, it shouldnt be!

  <If sdk="nextjs">
    ```tsx {{ filename: 'app/sso-setup/page.tsx' }}
    import { ConfigureSSO } from '@clerk/nextjs/experimental'

    export default function SSOSetupPage() {
      return <ConfigureSSO />
    }
    ```
  </If>

  <If sdk="react">
    ```tsx {{ filename: 'src/sso-setup.tsx' }}
    import { ConfigureSSO } from '@clerk/clerk-react/experimental'

    export default function SSOSetupPage() {
      return <ConfigureSSO />
    }
    ```
  </If>

  <If sdk="tanstack-react-start">
    ```tsx {{ filename: 'src/routes/sso-setup.tsx' }}
    import { ConfigureSSO } from '@clerk/tanstack-react-start/experimental'
    import { createFileRoute } from '@tanstack/react-router'

    export const Route = createFileRoute('/sso-setup')({
      component: SSOSetupPage,
    })

    function SSOSetupPage() {
      return <ConfigureSSO />
    }
    ```
  </If>

  <If sdk="astro">
    ```astro {{ filename: 'src/pages/sso-setup.astro' }}
    ---

>     import { __experimental_ConfigureSSO as ConfigureSSO } from '@clerk/astro/components'

Reviewer 2: We'll remove the `__experimental` prefix from here before releasing it GA - this is just used in the SDK now so we can incrementally implement internally

    The container `<div>` element with a rendered `<ConfigureSSO />` component instance.
  </Properties>

  #### <code>unmountConfigureSSO()</code> usage

  ```js {{ filename: 'main.js', mark: [19] }}
  import { Clerk } from '@clerk/clerk-js'

  // Initialize Clerk with your Clerk Publishable Key
  const clerkPubKey = import.meta.env.VITE_CLERK_PUBLISHABLE_KEY

  const clerk = new Clerk(clerkPubKey)
  await clerk.load()

  document.getElementById('app').innerHTML = `
    <div id="configure-sso"></div>
  `

  const configureSSODiv = document.getElementById('configure-sso')

  clerk.__experimental_mountConfigureSSO(configureSSODiv)

  // ...

  clerk.__experimental_unmountConfigureSSO(configureSSODiv)
  ```
</If>

## Properties

All props are optional.

<Properties>
  - `appearance?`
  - <code>[Appearance](/docs/guides/customizing-clerk/appearance-prop/overview) | undefined</code>

  Customization options to fully match the Clerk components to your own brand. These options serve as overrides and will be merged with the global `appearance` configuration (if one is provided). See the [`Appearance`](/docs/guides/customizing-clerk/appearance-prop/overview) docs for more information.

>   <If notSdk="js-frontend">

Reviewer 3: This makes the display appear weirdly - I can try to fix it but just wanted to clarify, the `fallback?` prop doesn't apply to JS SDK then? @im2nguyen 

<img width="717" height="542" alt="Image" src="https://github.com/user-attachments/assets/04a6b9d4-56ad-4d61-9edf-b4ebfff39fee" />

Author: yeah, it doesn't apply to js sdk but tagging @LauraBeatris to confirm
